src/sagas/nanoContract.js

```diff
@@ -215,16 +215,18 @@ function* registerNanoContractOnError(error) {
  * @param {string} req.before Transaction hash to start to get newer items
  * @param {string} req.after Transaction hash to start to get older items
  * @param {Object} req.wallet Wallet instance from redux state
+ * @param {boolean} req.useWalletService A flag that determines if wallet service is in use
  *
  * @returns {{
- *   history: {};
+ *   history: NcTxHistory;
  * }}
  *
  * @throws {Error} when request code is greater then 399 or when response's success is false
  */
 export async function fetchHistory(req) {
   const {
     wallet,
+    useWalletService,
```

Comment:
[note] Receive it from the saga effect is more efficient. More efficient than that would be a flag in the `wallet` instance to indicate the kind of wallet, wither `fullnode` or `walletService`. Do you think should add this flag in the wallet interface?